qt/applications/workbench/workbench/plotting/figureinteraction.py

```diff
@@ -96,6 +96,7 @@ def __init__(self, fig_manager):
         self._cids = []
         self._cids.append(canvas.mpl_connect("button_press_event", self.on_mouse_button_press))
         self._cids.append(canvas.mpl_connect("button_release_event", self.on_mouse_button_release))
+        self._cids.append(canvas.mpl_connect("button_release_event", self.on_button_release_legend_bounds_check))
```

Comment:
I think instead of adding a new connection, I would add a call to `on_button_release_legend_bounds_check` (maybe can just be called `legend_bounds_check`) to `on_mouse_button_release`.